realm/realm-library/src/objectServer/java/io/realm/mongodb/AppConfiguration.java

```diff
@@ -411,20 +412,26 @@ public void onError(SyncSession session, AppException error) {
                 }
             }
         };
-        private SyncClientResetStrategy defaultSyncClientResetStrategy = new DiscardUnsyncedChangesStrategy() {
+        private SyncClientResetStrategy defaultSyncClientResetStrategy = new RecoverOrDiscardUnsyncedChangesStrategy() {
```

Comment:
... or maybe align with the other SDKs and see what they are doing

Reply:
I believe it is sensible to move to the new recovery mode. I would add a breaking change. For the manual and discard local, the final result was the same, any unsynced data was lost. But now, it would be recovered.